x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -324,148 +336,181 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
-                        label={i18n.translate('xpack.lens.indexPatterns.toggleEmptyFieldsSwitch', {
-                          defaultMessage: 'Only show fields with data',
-                        })}
-                        data-test-subj="lnsEmptyFilter"
-                      />
-                    </EuiPopoverFooter>
-                  </EuiPopover>
-                }
-                clear={{
-                  title: i18n.translate('xpack.lens.indexPatterns.clearFiltersLabel', {
-                    defaultMessage: 'Clear name and type filters',
-                  }),
-                  'aria-label': i18n.translate('xpack.lens.indexPatterns.clearFiltersLabel', {
-                    defaultMessage: 'Clear name and type filters',
-                  }),
-                  onClick: () => {
+          <div className="lnsInnerIndexPatternDataPanel__filterWrapper">
+            <EuiFormControlLayout
+              icon="search"
+              clear={{
+                title: i18n.translate('xpack.lens.indexPatterns.clearFiltersLabel', {
+                  defaultMessage: 'Clear name and type filters',
+                }),
+                'aria-label': i18n.translate('xpack.lens.indexPatterns.clearFiltersLabel', {
+                  defaultMessage: 'Clear name and type filters',
+                }),
+                onClick: () => {
+                  setLocalState(s => ({
+                    ...s,
+                    nameFilter: '',
+                    typeFilter: [],
+                  }));
+                },
+              }}
+            >
+              <input
+                className="euiFieldText lnsInnerIndexPatternDataPanel__textField"
+                data-test-subj="lnsIndexPatternFieldSearch"
+                placeholder={i18n.translate('xpack.lens.indexPatterns.filterByNameLabel', {
+                  defaultMessage: 'Search for fields',
+                  description:
+                    'Search the list of fields in the index pattern for the provided text',
+                })}
+                value={localState.nameFilter}
+                onChange={e => {
+                  setLocalState({ ...localState, nameFilter: e.target.value });
+                }}
+                aria-label={i18n.translate('xpack.lens.indexPatterns.filterByNameAriaLabel', {
+                  defaultMessage: 'Search fields',
+                })}
+              />
+            </EuiFormControlLayout>
+          </div>
+          <div className="lnsInnerIndexPatternDataPanel__filtersWrapper">
+            <EuiPopover
+              id="dataPanelTypeFilter"
+              panelClassName="euiFilterGroup__popoverPanel"
+              panelPaddingSize="none"
+              anchorPosition="downLeft"
+              display="block"
+              isOpen={localState.isTypeFilterOpen}
+              closePopover={() => setLocalState(s => ({ ...localState, isTypeFilterOpen: false }))}
+              button={
+                <EuiFacetButton
```

Comment:
Maybe also due to the merge: When focussed, there is no padding to the left and the right of the count badge (background color ends directly with the element):
<img width="449" alt="Screenshot 2019-09-30 at 11 13 20" src="https://user-images.githubusercontent.com/1508364/65865345-8db98280-e373-11e9-9696-0ff76a177585.png">

Reply:
The alignments come straight from the component itself. I was able to fix the focus background color.